### ISSUE TITLE
Confusing parameter 'prefixed' to document GET endpoint

### ISSUE THREAD

**Issue by spl** [terminusdb/terminusdb#775]
**Describe the bug**

There is a boolean parameter called `prefixed` to the `GET /api/document` endpoint. From its name, I would guess that `prefixed=true` would add the appropriate `@base` or `@schema` prefix to things in the documents of the response. However, there are some confusing aspects of this that I don't understand:

* The default value in source is `true`:
   https://github.com/terminusdb/terminusdb/blob/249ea6b29f7aed949d2fa279f16e99db875914ff/src/server/routes.pl#L782
* The value of the `Prefixed` variable eventually finds its way to the `get_document` predicate as the `Compress` variable:
    https://github.com/terminusdb/terminusdb/blob/249ea6b29f7aed949d2fa279f16e99db875914ff/src/core/document/json.pl#L1545
* `Compress` seems to be used to “compress” URIs, which I presume means to add prefixes and is the opposite of what I'd expect for `prefixed`.
   https://github.com/terminusdb/terminusdb/blob/249ea6b29f7aed949d2fa279f16e99db875914ff/src/core/document/json.pl#L1559-L1561
* If I pass `prefixed=false&graph_type=instance`, I do indeed get prefixed fields, `@type`, and `@id`.
* If I pass `prefixed=false&graph_type=schema`, I do _not_ get any prefixed keys or values.

**Comment by spl** [comment]
> If I pass `prefixed=false&graph_type=schema`, I do _not_ get any prefixed keys or values.

Ignoring `prefixed` for schemas seems intentional. See `_Prefixed` below. Should it be documented like this, or should it be changed?

https://github.com/terminusdb/terminusdb/blob/249ea6b29f7aed949d2fa279f16e99db875914ff/src/core/api/api_document.pl#L75
https://github.com/terminusdb/terminusdb/blob/249ea6b29f7aed949d2fa279f16e99db875914ff/src/core/api/api_document.pl#L91
https://github.com/terminusdb/terminusdb/blob/249ea6b29f7aed949d2fa279f16e99db875914ff/src/core/api/api_document.pl#L133

**Comment by matko** [comment]
prefixed doesn't work for schemas yet. This is not intentional, it's a missing feature.

prefixed does mean compressed. we'll never add `@base` or `@schema` to things though, cause these are cases in which the prefix can be omitted. But if another prefix applies, such as when the user manually added a prefix `foo`, the compression could result in a `foo:Bar`.

**Comment by spl** [comment]
Thanks for the clarification, @matko. As we talked about elsewhere, I think the name `prefixed` is confusing, because, as you point out, the prefix is omitted if the prefix is `@base` or `@schema`. So, even though `prefixed=true`, not every ID is prefixed. I think compression more accurate describes what we're doing, but a name like `compress` is too ambiguous. What if we renamed the flag to `compress_ids`? We can still support `prefixed` but not document it.
